Plugins/GenerateDoccReference/GenerateDoccReference.swift

```diff
@@ -18,83 +21,101 @@ struct GenerateDoccReferencePlugin: CommandPlugin {
     context: PluginContext,
     arguments: [String]
   ) async throws {
-    // Locate generation tool.
     let generationTool = "generate-docc-reference"
     let generationToolFile = try context.tool(named: generationTool).path
-
-    // Create an extractor to extract plugin-only arguments from the `arguments`
-    // array.
     var extractor = ArgumentExtractor(arguments)
 
-    // Run generation tool once if help is requested.
-    if extractor.helpRequest() {
-      try generationToolFile.exec(arguments: ["--help"])
-      print(
-        """
-        ADDITIONAL OPTIONS:
-          --configuration <configuration>
-                      Tool build configuration used to generate the
-                      reference document. (default: release)
-
-        NOTE: The "GenerateDoccReference" plugin handles passing the "<tool>" and
-        "--output-directory <output-directory>" arguments. Manually supplying
-        these arguments will result in a runtime failure.
-        """)
+    if checkForHelpArgument(extractor: &extractor, generationToolFile: generationToolFile) {
       return
     }
 
-    // Extract configuration argument before making it to the
-    // "generate-docc-reference" tool.
     let configuration = try extractor.configuration()
+    let buildArtifacts = try buildPackage(configuration: configuration)
+
+    try generateDocs(context: context, with: extractor, buildArtifacts: buildArtifacts, generationToolFile: generationToolFile)
+  }
 
-    // Build all products first.
-    print("Building package in \(configuration) mode...")
+  private func buildPackage(configuration: PackageManager.BuildConfiguration) throws -> [PackageManager.BuildResult.BuiltArtifact] {
+    // The package must be built before documentation can be generated
+    // because DocC requires compiled executables to extract symbols.
```

Comment:
I appreciate the focus on "why" rather than "what" but this is not correct.

Not only is DocC not the tool that extracts symbols but there are this command doesn't extract any symbols (in the meaning that symbols has in DocC).

The reason why the executable needs to build is that the `generate-docc-reference` executable runs the tool and passes the `--experimental-dump-help` flag to extract the information about the command line interface's structure.